File: Java/Solutions/4.2. Complex Conditions - Exam Problems/Trip.java
import java.lang.*;
import java.math.BigDecimal;
import java.text.DecimalFormat;
import java.util.Scanner;

public class Trip {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        // Read input data
        BigDecimal budget = new BigDecimal(scanner.nextLine());
        String season = scanner.nextLine();

        // Initialization of the required parameters
        String destinationResult = "";
        String holidayInformation = "";
        BigDecimal moneySpent = new BigDecimal("0.00");

        DecimalFormat formatter = new DecimalFormat("0.00");
        // Calculate result
        if (budget.compareTo(BigDecimal.valueOf(100.00)) <= 0) {
            destinationResult = "Bulgaria";
            if (season.equals("summer")) {
                moneySpent = BigDecimal.valueOf(0.30).multiply(budget);
                holidayInformation = String.format("Camp - %s", formatter.format(moneySpent));
            } else {
                moneySpent = BigDecimal.valueOf(0.70).multiply(budget);
                holidayInformation = String.format("Hotel - %s", formatter.format(moneySpent));
            }
        } else if (budget.compareTo(BigDecimal.valueOf(1000.00)) <= 0) {
            destinationResult = "Balkans";
            if (season.equals("summer")) {
                moneySpent = BigDecimal.valueOf(0.40).multiply(budget);
                holidayInformation = String.format("Camp - %s", formatter.format(moneySpent));
            } else {
                moneySpent = BigDecimal.valueOf(0.80).multiply(budget);
                holidayInformation = String.format("Hotel - %s", formatter.format(moneySpent));
            }
        } else {
            destinationResult = "Europe";
            moneySpent = BigDecimal.valueOf(0.90).multiply(budget);
            holidayInformation = String.format("Hotel - %s", formatter.format(moneySpent));
        }

        System.out.println(String.format("Somewhere in %s", destinationResult));
        System.out.println(holidayInformation);
    }
}

File: CSharp/Solutions/4.2. Complex Conditions - Exam Problems/Trip.cs
namespace Trip
{
    using System;

    public static class Trip
    {
        public static void Main()
        {
            // Read input data
            decimal budget = decimal.Parse(Console.ReadLine());
            string season = Console.ReadLine();

            // Initialization of the required parameters
            string destinationResult = string.Empty;
            string holidayInformation = string.Empty;
            decimal moneySpent = 0.00M;

            // Calculate result
            if (budget <= 100.00M)
            {
                destinationResult = "Bulgaria";
                if (season.Equals("summer"))
                {
                    moneySpent = 0.30M * budget;
                    holidayInformation = string.Format("Camp - {0:F2}", moneySpent);
                }
                else
                {
                    moneySpent = 0.70M * budget;
                    holidayInformation = string.Format("Hotel - {0:F2}", moneySpent);
                }
            }
            else if (budget <= 1000.00M)
            {
                destinationResult = "Balkans";
                if (season.Equals("summer"))
                {
                    moneySpent = 0.40M * budget;
                    holidayInformation = string.Format("Camp - {0:F2}", moneySpent);
                }
                else
                {
                    moneySpent = 0.80M * budget;
                    holidayInformation = string.Format("Hotel - {0:F2}", moneySpent);
                }
            }
            else
            {
                destinationResult = "Europe";
                moneySpent = 0.90M * budget;
                holidayInformation = string.Format("Hotel - {0:F2}", moneySpent);
            }

            Console.WriteLine("Somewhere in {0}", destinationResult);
            Console.WriteLine(holidayInformation);
        }
    }
}
